File: App/torneio_app/lib/core/models/checklist_torneio_item.dart
class ChecklistTorneioItem {
  final String id;
  final String torneioId;
  final String item;
  final DateTime? data;
  final String? responsavel;
  final bool concluido;

  const ChecklistTorneioItem({
    required this.id,
    required this.torneioId,
    required this.item,
    this.data,
    this.responsavel,
    required this.concluido,
  });

  factory ChecklistTorneioItem.fromJson(Map<String, dynamic> json) => ChecklistTorneioItem(
        id: json['id'] as String,
        torneioId: json['torneioId'] as String,
        item: json['item'] as String? ?? '',
        data: json['data'] != null ? DateTime.tryParse(json['data'] as String) : null,
        responsavel: json['responsavel'] as String?,
        concluido: json['concluido'] as bool? ?? false,
      );
}

File: Retaguarda/Torneio.Domain/Entities/ChecklistTorneioItem.cs
namespace Torneio.Domain.Entities;

public class ChecklistTorneioItem
{
    public Guid Id { get; private set; }
    public Guid TorneioId { get; private set; }
    public string Item { get; private set; } = null!;
    public DateTime? Data { get; private set; }
    public string? Responsavel { get; private set; }
    public bool Concluido { get; private set; }

    private ChecklistTorneioItem() { }

    public static ChecklistTorneioItem Criar(
        Guid torneioId,
        string item,
        DateTime? data,
        string? responsavel,
        bool concluido)
    {
        return new ChecklistTorneioItem
        {
            Id = Guid.NewGuid(),
            TorneioId = torneioId,
            Item = item,
            Data = data,
            Responsavel = string.IsNullOrWhiteSpace(responsavel) ? null : responsavel.Trim(),
            Concluido = concluido
        };
    }

    public void Atualizar(string item, DateTime? data, string? responsavel, bool concluido)
    {
        Item = item;
        Data = data;
        Responsavel = string.IsNullOrWhiteSpace(responsavel) ? null : responsavel.Trim();
        Concluido = concluido;
    }
}
